docs/machine-learning/Long_Short_Term_Memory.md

```diff
@@ -0,0 +1,162 @@
+
+2. **Bidirectional LSTM (BiLSTM)**:  
+   This architecture consists of two LSTM layers that process the input in both forward and backward directions. It is especially useful for tasks where future context is as important as past context.
+
+3. **Stacked LSTM**:  
+   Multiple LSTM layers are stacked on top of each other, enabling the network to learn more complex and hierarchical patterns in the data.
+
+4. **Peephole LSTM**:  
+   In Peephole LSTMs, the gates also receive input from the cell state, allowing them to "peek" into the cell state during updates.
+
+### Problem Statement:
+Given sequential data such as time-series, text, or audio, the goal of an LSTM is to make accurate predictions by effectively capturing long-term dependencies while mitigating issues like vanishing gradients.
+
+### Key Concepts:
+- **Memory Cell**:  
+  LSTMs maintain a memory cell that persists over time and can store relevant information while discarding irrelevant information through its gates.
+
+- **Vanishing Gradient Problem**:  
+  LSTMs help mitigate the vanishing gradient problem in vanilla RNNs, which makes them better suited for learning long-term dependencies.
+
+- **Gates**:  
+  LSTMs employ three main gates (forget, input, and output gates) to regulate the flow of information, ensuring that important information is remembered and unimportant information is forgotten.
+
+- **Backpropagation Through Time (BPTT)**:  
+  LSTMs use BPTT to update weights by propagating gradients back through the entire sequence. Due to the gating mechanism, LSTMs avoid the vanishing gradient problem better than traditional RNNs.
+
+### Steps Involved:
+1. **Input Data**:  
+   Sequential data is fed into the LSTM one time step at a time.
+
+2. **Gate Calculations**:  
+   The forget, input, and output gates compute which information to retain, update, and output at each time step.
+
+3. **Cell State Update**:  
+   The cell state is updated using the information from the forget and input gates, deciding which information to keep or discard.
+
+4. **Hidden State Calculation**:  
+   The output gate computes the hidden state, which is used as the output for the current time step and passed to the next LSTM cell.
+
+5. **Final Output**:  
+   After processing the entire sequence, the final hidden state can be used for tasks like sequence classification, or intermediate outputs can be used for tasks like sequence generation.
+
+### Split Criteria:
+LSTMs process data in sequential order, using the previous hidden state and cell state at each time step to inform the current step’s calculations. They are particularly useful for tasks where maintaining information over long sequences is critical.
+
+### Time Complexity:
+- **Training Complexity**:  
+  The time complexity of training an LSTM is proportional to the sequence length \( T \), the size of the input \( n \), and the number of hidden units \( h \), and is approximately \( O(T \cdot n \cdot h^2) \).
+
```

Comment:
mathematical formula using LaTeX 

for example
`\( T \)` $\rightarrow$ `$T$`

and apply `$` for all